### PR TITLE
[utfcpp] Update to 4.0.2

### PR DESCRIPTION
**Pull request by RT2Code**
Update utfcpp from 3.2.5 to 4.0.2 : 
https://github.com/nemtrif/utfcpp/releases/tag/v4.0.1
https://github.com/nemtrif/utfcpp/releases/tag/v4.0.2

The CMakeLists.txt file was [entirely rewritten](https://github.com/nemtrif/utfcpp/pull/110/commits/72527bda4b26b804d9e7569541b400ef1749e604) on the version 4.0.0, and I removed the CMake options and the MinGW fix because they aren't necessary anymore. The assimp port also required some minor tweaks to be able to build with this new version.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by FrankXie05** [comment]
@RT2Code `VTK` also depends on `utfcpp`.
https://github.com/microsoft/vcpkg/blob/01acfdfde3ed99280d3883a8fccd5fa4408f5214/ports/vtk/vcpkg.json#L49

**Comment by RT2Code** [comment]
Yes, I also tested it when I made this PR, and it still builds fine without any change.

**Review by FrankXie05 — APPROVED** [review]

File: ports/assimp/build_fixes.patch [hunk 1]
```diff
@@ -171,7 +171,7 @@ index ca37ba1..2ce7bdd 100644
  #  include <utf8.h>
  #else
 -#  include "../contrib/utf8cpp/source/utf8.h"
-+#  include <utf8.h>
++#  include <utf8cpp/utf8.h>
```
Comment (Osyotr) [inline ports/assimp/build_fixes.patch:174]: Why is this needed if `include/utf8cpp` is in include path?

File: ports/utfcpp/fix-include-path.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 2ca50a6..d707dc7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -10,7 +10,7 @@ include(GNUInstallDirs)
+ 
+ target_include_directories(utf8cpp INTERFACE
+     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source>"
+-    $<INSTALL_INTERFACE:include>
++    $<INSTALL_INTERFACE:include/utf8cpp>
```
Comment (Osyotr) [inline ports/utfcpp/fix-include-path.patch:10]: What is the purpose of this patch?
